src/ossia/network/base/node_functions.cpp

```diff
@@ -638,16 +641,59 @@ std::vector<ossia::net::node_base*> list_all_child(ossia::net::node_base* node)
     return ossia::net::get_priority(*n1) > ossia::net::get_priority(*n2);
   });
 
+  int next_depth = -1;
+  if(depth > 0)
+    next_depth = depth - 1;
+
   for (auto it = children.begin(); it != children.end(); it++)
   {
     list.push_back(*it);
-    auto nested_list = list_all_child(*it);
-    list.insert(list.end(), nested_list.begin(), nested_list.end());
+    if(next_depth != 0)
+    {
+      auto nested_list = list_all_children(*it, next_depth);
+      list.insert(list.end(), nested_list.begin(), nested_list.end());
+    }
   }
 
   return list;
 }
 
+/**
+ * @brief fuzzysearch: search for nodes that match the pattern string
+ * @param nodes: vector of nodes from where to start
+ * @param pattern: strings to search
+ * @return a vector of fuzzysearch_result sorted in descending score order
+ */
+void fuzzysearch(std::vector<ossia::net::node_base*> nodes,
+                 const std::vector<std::string>& patterns,
+                 std::vector<fuzzysearch_result>& results)
+{
+  results.clear();
+
+  for(const auto& node : nodes)
+  {
+    auto children = list_all_children(node);
+
+    results.reserve(results.size() + children.size());
+
+#pragma omp parallel for
+    for(const auto& n : children)
+    {
+      std::string oscaddress = ossia::net::osc_parameter_string_with_device(*n);
+      double percent = 1.0;
+      for(const auto& pattern : patterns)
+      {
+        percent *= rapidfuzz::fuzz::partial_ratio(oscaddress, pattern) / 100.;
+      }
+      results.push_back({percent * 100., oscaddress, n});
+    }
+
+    ossia::sort(results, [](const fuzzysearch_result& left, const fuzzysearch_result& right){
```

Comment:
for the future: C++20 will have a neat way to do that without needing a long lambda: 
`ranges::sort(particles, std::greater{}, &fuzzysearch_result::score);`

Reply:
e.g. https://en.cppreference.com/w/cpp/algorithm/ranges/sort